### ISSUE TITLE
Matrix *= operator does assignment instead of multiplication

### ISSUE THREAD

**Issue by dkorpel** [d-gamedev-team/gfm#206]
Doing `matrix *= 2` seems to be the same as `matrix = 2`:
```D
import gfm.math.matrix: mat4f;
mat4f mvp = mat4f.identity;
writeln(mvp);
mvp *= 2;
writeln(mvp);
```
Output:
```
[1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1]
[2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
```
This is quite unintuitive behavior. `+=` and `-=` work as expected, `/=` doesn't compile. So maybe `*=` shouldn't too? Or it should do element-wise multiplication I'd say.

**Comment by p0nce** [comment]
What happens is that 2 is promoted to `mat4f(2)` which is 2 everywhere. 
I agree it should yield `[2, 0, 0, 0, 0, 2, 0, 0, 0, 0, 2, 0, 0, 0, 0, 2]`.
Good catch.

**Comment by p0nce** [comment]
Besides if we fix this, matrix mult is much more expensive than by-elem mult.

**Comment by p0nce** [comment]
Fixed in v7.0.7  , `*` and `*=` will now work similarly.
